### PR TITLE
[release-4.12] Revert extra private key secret creation

### PR DESCRIPTION
**Pull request by jrvaldes**
The `azure-e2e-operator` job is being failing [1] due to a timing problem while testing the private key secret change during the creation test suite. The later was fixed in 09c2bfb4589cefe4d3cb7b646bd369ee272f1613 by introducing a  wait after changing the private key secret on Azure to check all machines provisioned with the outdated windows-user-data were deleted. 

In 69e56dff8b68e68edb78a909f7f51ee6c0c9ab1d the private key secret change test was incorrectly  re-introduced making the timing problem resurface. This change removes the re-introduced code that triggers the private key secret change to fix the timing problem shown in the azure-e2e-operator job.

The following jobs shows successful runs of the azure-e2e-operator job as validation of the changes introduced here.
- [release-4.12-azure-e2e-operator #1745196721991847936](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_windows-machine-config-operator/2002/pull-ci-openshift-windows-machine-config-operator-release-4.12-azure-e2e-operator/1745196721991847936)
- [release-4.12-azure-e2e-operator #1745264177334718464](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_windows-machine-config-operator/2002/pull-ci-openshift-windows-machine-config-operator-release-4.12-azure-e2e-operator/1745264177334718464)

[1] Screenshot of the test results as of 204-01-12 of the azure-e2e-operator job for release-4.12 branch
image

Source: https://prow.ci.openshift.org/job-history/gs/test-platform-results/pr-logs/directory/pull-ci-openshift-windows-machine-config-operator-release-4.12-azure-e2e-operator

Blocks:
- https://github.com/openshift/windows-machine-config-operator/pull/1994
- https://github.com/openshift/windows-machine-config-operator/pull/2012
- https://github.com/openshift/windows-machine-config-operator/pull/1932

### CODE REVIEW

**Comment by jrvaldes** [comment]
azure-e2e-operator   Job [succeeded](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_windows-machine-config-operator/2002/pull-ci-openshift-windows-machine-config-operator-release-4.12-azure-e2e-operator/1745196721991847936).

**Comment by mansikulkarni96** [comment]
Thanks for fixing this!
/approve

File: test/e2e/create_test.go [hunk 1]
```diff
@@ -165,11 +165,6 @@ func (tc *testContext) testMachineConfiguration(t *testing.T) {
 
 	machines, err := tc.waitForWindowsMachines(int(gc.numberOfMachineNodes), "Provisioned", false)
 	require.NoError(t, err, "error waiting for Windows Machines to be provisioned")
-	if tc.CloudProvider.GetType() == config.AzurePlatformType {
```
Comment (sebsoto) [inline test/e2e/create_test.go:168]: Without having full context of both commits you linked to, its not possible to understand why this change is being made.
Can you write the commit message in a way that states what this change is doing on its own? Think about someone looking back in commit history trying to figure out why this change was made.

**Review by sebsoto — CHANGES_REQUESTED** [review]

File: test/e2e/create_test.go (hunk 1 above)
Comment (jrvaldes) [reply test/e2e/create_test.go:168]: Does https://github.com/openshift/windows-machine-config-operator/pull/2002#issue-2075240715 suffice?

Comment (sebsoto) [reply test/e2e/create_test.go:168]: I'd like to see it within the commit message saying this is already being tested somewhere, and where

Comment (jrvaldes) [reply test/e2e/create_test.go:168]: @sebsoto  I updated the description of the PR with the supporting information you are requesting. PTAL https://github.com/openshift/windows-machine-config-operator/pull/2002#issue-2075240715.

The PR description is accessible from the commit history and will help with context in case we need to revisit this. I leaned towards not pushing an update to avoid triggering all CI jobs again considering this PR is blocking other PRs.
